### PR TITLE
Update unit/mock testing

### PR DESCRIPTION
**Pull request by simplynaveen20**
Update unit/mock testing

## Closing issues

#477

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs [hunk 1]
```diff
@@ -42,6 +42,30 @@ public ContainerResponse()
         /// </summary>
         public virtual Container Container { get; private set; }
 
+        /// <inheritdoc/>
+        public override Headers Headers { get; protected set; }
+
+        /// <inheritdoc/>
+        public override ContainerProperties Resource { get; protected set; }
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs]: Can we remove protected set? I don't see any reason for it.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs [hunk 2]
```diff
@@ -42,6 +42,30 @@ public ContainerResponse()
         /// </summary>
         public virtual Container Container { get; private set; }
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs]: Can we remove the private set so it's readonly?

File: Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedResponse.cs [hunk 3]
```diff
@@ -30,13 +31,36 @@ public FeedResponse()
...
+        public override Headers Headers { get; protected set; }
+
+        /// <inheritdoc/>
+        public override IEnumerable<T> Resource { get; protected set; }
+
+        /// <inheritdoc/>
+        public override HttpStatusCode StatusCode { get; }
+
+        /// <inheritdoc/>
+        public override double RequestCharge => this.Headers.RequestCharge;
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedResponse.cs]: What happens if headers is null? Someone creates it with an empty constructor?

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs (hunk 1 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs]: Actually it came from Resource class as one of the child (QueryResource) was throwing on set , so i had to override on all children. Removing unnecessary throw on set and removing it from all the responses.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs (hunk 2 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerResponse.cs]: Container and Database had this unnecessary private , removing it.

File: Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedResponse.cs (hunk 3 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedResponse.cs]: Empty constructor is for mocking purpose only , that being said , i am moving all public to protected.
Secondly null check anyway needed on Header for all the resources which was missing earlier, so adding it now.

File: Microsoft.Azure.Cosmos/src/Resource/Response.cs [hunk 4]
```diff
@@ -15,35 +14,19 @@ public abstract class Response<T>
         /// <summary>
         /// Create an empty cosmos response for mock testing
         /// </summary>
-        public Response()
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Response.cs]: Remove

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Response.cs]: Done

**Pull request merged**